internal/zero/cmd/import_ui.go

```diff
@@ -0,0 +1,553 @@
+		"envoy_bind_config_freebind":             {Unsupported: true, Note: noteFeatureNotYetAvailable},
+		"envoy_bind_config_source_address":       {Unsupported: true, Note: noteFeatureNotYetAvailable},
+		"google_cloud_serverless_authentication_service_account": {Secret: true},
+		"idp_client_secret":                    {Secret: true},
+		"databroker_storage_connection_string": {Secret: true},
+		"metrics_certificate":                  {Unsupported: true, Note: noteFeatureNotYetAvailable},
+		"metrics_client_ca":                    {Unsupported: true, Note: noteFeatureNotYetAvailable},
+		// "metrics_certificate":                  {Note: noteCertificate(1)},
+		// "metrics_client_ca":                    {Note: noteCertificate(1)},
+		"certificate_authority": {Note: noteCertificate(1)},
+		"certificates":          {Note: noteCertificate(len(cfg.GetSettings().GetCertificates()))},
+		"downstream_mtls.crl":   {Unsupported: true, Note: noteFeatureNotYetAvailable},
+		"downstream_mtls.ca":    {Note: noteCertificate(1)},
+	}
+	if dm := cfg.GetSettings().GetDownstreamMtls(); dm != nil {
+		if dm.Enforcement != nil {
+			switch *dm.Enforcement {
+			case configpb.MtlsEnforcementMode_POLICY:
+			case configpb.MtlsEnforcementMode_POLICY_WITH_DEFAULT_DENY:
+			case configpb.MtlsEnforcementMode_REJECT_CONNECTION:
+				// this is a special case - zero does not support this mode, but we cannot continue
+				// with a partial import because it fundamentally changes the behavior of all routes
+				// and policies in the system
+				log.Fatal().Msg("downstream mtls enforcement mode 'reject_connection' is not supported")
+			}
+		}
+	}
+	if cfg.GetSettings().GetServices() != "all" {
+		m["services"] = ImportHints{Ignored: true, Note: `only "all" is supported`}
+	}
+	if cfg.GetSettings().GetCodecType() != http_connection_managerv3.HttpConnectionManager_AUTO {
+		m["codec_type"] = ImportHints{Ignored: true, Note: `only "auto" is supported`}
+	}
+	return m
+}
+
+type ImportUI struct {
+	form             *huh.Form
+	selectedSettings []string
+	selectedRoutes   []string
+}
+
+func NewImportUI(cfg *configpb.Config, quotas *cluster_api.ConfigQuotas) *ImportUI {
+	settingsImportHints := computeSettingsImportHints(cfg)
+
+	presentSettings := fieldmasks.Leaves(
+		fieldmasks.Diff(
+			config.NewDefaultOptions().ToProto().GetSettings().ProtoReflect(),
+			cfg.GetSettings().ProtoReflect(),
+		),
+		cfg.Settings.ProtoReflect().Descriptor(),
+	)
+	slices.Sort(presentSettings.Paths)
+	settingsOptions := huh.NewOptions(presentSettings.Paths...)
+
+	ui := &ImportUI{
+		selectedSettings: slices.Clone(presentSettings.Paths),
+	}
+
+	for i, value := range presentSettings.Paths {
+		if hints, ok := settingsImportHints[value]; ok {
+			switch {
+			case hints.Ignored:
+				note := ""
+				if hints.Note != "" {
+					note = fmt.Sprintf(": %s", hints.Note)
+				}
+				settingsOptions[i].Key = fmt.Sprintf("\x1b[9m%s\x1b[29m \x1b[2m(ignored%s)\x1b[22m", settingsOptions[i].Key, note)
```

Comment:
Are these ANSI escape codes for color formatting? Is that also what the lipgloss package does? Is there a reason for some of these styles to be written directly like this and some to be added via the lipgloss package?

Reply:
For some reason, the lipgloss strikethrough renderer ends the string with `\x1b[0m`  which is the sequence for "clear all formatting". This interferes with the color for selected/unselected fields, which is handled inside this library, not by us. Instead we use `\x1b[29m` which turns off strikethrough but doesn't change anything else.

Reply:
I see there's a `ColorProfile` / `EnvNoColor()` check below — do we need to make the escape codes here conditional too, to match whether color is enabled?

(I'm totally unfamiliar with some of these libraries, so please bear with me if this is a dumb question.)

Reply:
Good question. That code will force ANSI 16 colors if we otherwise can't determine what the terminal supports, since the default is no colors. This UI is mostly intended to be used from inside a container via interactive docker/kubectl exec, but those environments usually won't have the environment variables like TERM/COLORTERM that are used to detect terminal color support.